apps/api/models/appointment.js

```
@@ -233,6 +238,41 @@ export const addPatientToClass = async ({ patientId, appointmentId }) => {
   }))[0]
 }
 
+export const authorizeClient = async (userId, apptId) => {
+  const appt = await cliniko.findOne({
+    resource: 'individual_appointments',
+    id: apptId
+  })
+  const appPatient = await cliniko.findOne({
+    resource: 'patients',
+    url: appt.patient.links.self
+  })
+  const user = await auth0.getUser({ id: userId })
+
+  if (user.user_metadata.patientId !== appPatient.id) {
+    throw new Error('Unauthorized')
+  }
+}
+
+const updateAppointment = async (_root, args, ctx) => {
+  await authorizeClient(ctx.user.id, args.appointmentId)
+  let appt
+  try {
+    appt = await cliniko.update({
+      resource: 'appointments',
+      id: args.appointmentId,
+      data: {
+        starts_at: args.startTime,
+        // Null value for 'ends_at' will calculate the value according to ApptType duration
+        ends_at: null
+      }
+    })
+  } catch (err) {
+    throw new Error(err)
```

Sorry, I didn't see this earlier, but this `try/catch` is unnecessary. Notice how you're throwing the exact same error you caught.`try` simply prevents an error within its braces from "throw"ing that procedure.

The only reason you would use try/catch is if you were going to do something different with the `err` than throw it right back (such as ignoring the error and logging it, transforming the error into a message appropriate to surface to the user, etc.). Most often this code lives "at the root" of your function calls (e.g. in our [jobs runner](https://github.com/findOctave/cortex/blob/master/apps/api/jobs/cli.js#L7)) or when something exceptional happens that we want to uniquely handle on the spot (such as [emailing people when a charge fails](https://github.com/findOctave/cortex/blob/master/apps/api/jobs/charge-sessions/network.js#L296)). When in doubt, just leave `try`/`catch` out, let the function throw, and let the error bubble up to one of these handlers.